File: jscomp/test/js_dom_test.ml

let suites = Mt.[
  (* These aren't available on node, so this is just to make sure it compiles *)

  ("Dom sanity check", (fun _ -> 
    ThrowAny(fun () -> Dom.Storage.(local |> get "key") |> ignore);
  ));

  ("Js.Dom sanity check", (fun _ -> 
    ThrowAny(fun () -> Js.Dom.Storage.(local |> get "key") |> ignore);
  ));
]

;; Mt.from_pair_suites __FILE__ suites

File: jscomp/test/js_dom_test.js
'use strict';

var Mt           = require("./mt.js");
var Block        = require("../../lib/js/block.js");
var Js_primitive = require("../../lib/js/js_primitive.js");

var suites_000 = /* tuple */[
  "Dom sanity check",
  (function () {
      return /* ThrowAny */Block.__(7, [(function () {
                    Js_primitive.null_to_opt(localStorage.getItem("key"));
                    return /* () */0;
                  })]);
    })
];

var suites_001 = /* :: */[
  /* tuple */[
    "Js.Dom sanity check",
    (function () {
        return /* ThrowAny */Block.__(7, [(function () {
                      Js_primitive.null_to_opt(localStorage.getItem("key"));
                      return /* () */0;
                    })]);
      })
  ],
  /* [] */0
];

var suites = /* :: */[
  suites_000,
  suites_001
];

Mt.from_pair_suites("js_dom_test.ml", suites);

exports.suites = suites;
/*  Not a pure module */
